feat: Get more UI working

dtribble opened the pull request:
- make `address` a parameter
- better factor the main window
- use a store for the in-progress offer
- show received 2nd party invites
- button to setup the first and second offers
- enable second offer if invite is present

Note that "Make Second Offer" is different because it's making a "Purse" offer rather than a "Contract" offer (which determines where the invitation is found).

After "Connect Wallet"
<img width="822" alt="image" src="https://github.com/agoric-labs/dapp-swaparoo/assets/739213/248cabae-6458-4667-8b1e-81066f3b17d6">

After "First Offer"
<img width="822" alt="image" src="https://github.com/agoric-labs/dapp-swaparoo/assets/739213/82296687-bc32-4a37-84c2-3558ac49b816">

After "Make Offer"
<img width="822" alt="image" src="https://github.com/agoric-labs/dapp-swaparoo/assets/739213/592165b1-e89f-48e3-9759-795725727124">

As user "swappa"
<img width="1205" alt="image" src="https://github.com/agoric-labs/dapp-swaparoo/assets/739213/eb99c2b2-e1e4-412e-898c-3dbc7b9058b4">

After "Match First", then "Make Second Offer"
<img width="1205" alt="image" src="https://github.com/agoric-labs/dapp-swaparoo/assets/739213/f89b8536-8e48-48bb-8d32-589dcdaadd96">

Success
<img width="768" alt="image" src="https://github.com/agoric-labs/dapp-swaparoo/assets/739213/3595c723-3779-4e3f-993f-f1277797050e">

<img width="768" alt="image" src="https://github.com/agoric-labs/dapp-swaparoo/assets/739213/e90b7bf0-c92f-487d-88c2-b1b823993ddc">

dckc committed:
build: add a target to `agoric install` inside the container (WIP)

work-around for bundle size problem
https://github.com/Agoric/agoric-sdk/issues/8621#issuecomment-1868590526

dtribble committed:
See invitations in UI (badly)

dtribble committed:
feat: detect incoming swap offers

dtribble committed:
feat: show received invites

dtribble committed:
feat: hack to get match offers working

dckc commented:
4e655ea turned out to be not necessary; better punt that one

dckc commented:
@dtribble if you've got this running, would you add a screenshot or two to this PR?
Sam usually does that, and I find it really helpful.

dtribble commented:
> @dtribble if you've got this running, would you add a screenshot or two to this PR? Sam usually does that, and I find it really helpful.

Yes. In that case I'll make a few more changes so they are coherent and update the PR.

dtribble committed:
feat: get full demo working

- button to setup the first and second offers
- enable second offer if invite is present
- use store for offer fields

dtribble committed:
fix: line deleted accidentally

samsiegart commented on ui/src/App.tsx:
```diff
@@ -14,12 +14,13 @@ import {
 export const contractName = 'swaparoo';
-const gov1 = 'agoric14t2eagaphnkj33l9hvcrf90t3xffkt0u3xy6uh';
+// const gov1 = 'agoric14t2eagaphnkj33l9hvcrf90t3xffkt0u3xy6uh';
+const swappa = 'agoric13whlmf7akvmg05n3zx37v7r0htyasa4m28j9cu';
```
Can we just do `const recipientAddr = agoric13whlmf7akvmg05n3zx37v7r0htyasa4m28j9cu`? This variable name is a bit cryptic and seems redundant

samsiegart commented on ui/src/App.tsx:
```diff
@@ -46,11 +54,26 @@ interface Purse {
+  want?: object;
+}
+
+const stringify = (arg) => {
```
I know this is just to get things working but maybe a TODO here to display amounts better would be good. Looks like it's intended for displaying non-NAT values? https://github.com/Agoric/wallet-app/blob/main/wallet/src/components/PurseValue.tsx is a good reference, though I'd prefer not to use material UI as it does.

samsiegart commented on ui/src/App.tsx:
```diff
@@ -80,12 +103,27 @@ const setup = async () => {
   );
 };
 
+const Banner = () => (
```
This probably isn't necessary, it would free up some real estate if we just removed it

samsiegart commented on ui/src/App.tsx:
```diff
@@ -138,101 +230,150 @@ const makeOffer = () => {
+        </label>
+
+        <br />
+        {/* // select give asset
```
Maybe a `TODO: ` here?

samsiegart commented on ui/src/App.tsx:
```diff
@@ -138,101 +230,150 @@ const makeOffer = () => {
+
+const App = () => {
+  useEffect(() => {
+    setup();
```
Could this be renamed to `watchContract` or something? Trying to reason about, "what does `setup` do?"

samsiegart commented on ui/src/App.tsx:
```diff
@@ -98,32 +136,86 @@ const connectWallet = async () => {
       source: 'contract',
       instance: contractInstance,
       publicInvitationMaker: 'makeFirstInvitation',
       // HACK setup a trade
```
This comment doesn't explain much, maybe give an idea of what would ideally be included in `invitationArgs`?

samsiegart commented on ui/src/App.tsx:
```diff
@@ -138,101 +230,150 @@ const makeOffer = () => {
-        <button onClick={onClick}>{buttonLabel}</button>
+      </div>
+      <div style={{ textAlign: 'left' }}>
+        <h4  >GIVE </h4>
```
Nit: remove empty space after `h4`

Looks like we have prettier set up already, but no `.workflow` in this repo to use it. Eventually we can add something like https://github.com/Agoric/dapp-psm/blob/main/.github/workflows/test.yml

Also, maybe something in the README to suggest the prettier extension in vscode to get auto-format-on-save

samsiegart reviewed (approved):
Cool to see this working! Left some non-blocking suggestions, mostly focused around code clarity and future improvements

dckc reviewed (commented):
I defer to @samsiegart 

_submitting this review to get it off my queue_

samsiegart commented:
At this point we have swaparoo functionality in dapp-agoric-basics, can we abandon this?

dckc commented:
yes, I think so.
